### PR TITLE
City Dialog: ExpCol present units list

### LINKED ISSUE #2295: Add "expand" button to units panel in new City Dialog
**Is your feature request related to a problem? Please describe.**
When you have a large number of units in a city, it can be difficult to see them and even harder to find and pick the one you want to engage the enemy with.

**Describe the solution you'd like**
Reduce the units panel to only show a single row of units (not the 1.5 we have now) and then when there is more than one row enable an "up arrow" expand button that would enlarge the panel to something like 4 or 5 rows max to make it easier to see/work with the units.

**Describe alternatives you've considered**
None other than the status quo. We do have grand plans to make more changes to the city dialog, but that is probably a long way off.

**Additional context**
Came up in a conversation with @hugomflavio

### PR DESCRIPTION
**Pull request by jwrober**
Closes #2295

### CODE REVIEW

File: client/citydlg.cpp [hunk 1]
```diff
@@ -1789,11 +1791,37 @@ void city_dialog::update_units()
     units = pcity->tile->units;
   }
 
+  // set direction of the expand/collapse arrow
+  if (present_units_exp == false) {
```
Comment (lmoureaux) [inline client/citydlg.cpp:1795]:
```suggestion
  if (!present_units_exp) {
```

File: client/citydlg.cpp [hunk 2]
```diff
@@ -1789,11 +1791,37 @@ void city_dialog::update_units()
...
-  ui.curr_units->setText(QString(buf));
+  ui.present_units_label->setText(QString(buf));
+}
+
+/**
+ * Slot to expand or collapse the present units list
+ */
+void city_dialog::present_units_exp_col()
+{
+  if (present_units_exp == false) {
```
Comment (lmoureaux) [inline client/citydlg.cpp:1814]:
```suggestion
  if (!present_units_exp) {
```

**Review by lmoureaux — COMMENTED** [review]
Small style comments. IMNSHO the conditions would be easier to read if the `if` and `else` cases were inverted

**Comment by jwrober** [comment]
> Small style comments. IMNSHO the conditions would be easier to read if the `if` and `else` cases were inverted

so do the true condition first with else being false?

**Comment by lmoureaux** [comment]
> > Small style comments. IMNSHO the conditions would be easier to read if the `if` and `else` cases were inverted
> 
> so do the true condition first with else being false?

Yes, to avoid having to think about the `!` or `== false`. But it's eventually a matter of taste.

**Review by hugomflavio — COMMENTED** [review]
tested, looks pretty cool!

**Pull request merged**